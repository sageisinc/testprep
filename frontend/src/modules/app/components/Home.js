import React, { useState, useEffect } from "react";
import "./App.css";
import { config } from "../../../config/constants.js";
import { Link } from "react-router-dom";

function rot13(s) {
  return s.replace(
    /[A-Z]/gi,
    (c) =>
      "NOPQRSTUVWXYZABCDEFGHIJKLMnopqrstuvwxyzabcdefghijklm"[
        "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz".indexOf(c)
      ]
  );
}

const Home = () => {
  const [message, setMessage] = useState("");
  const [someText, setSomeText] = useState("Sreenzragnf");
  const [answer, setAnswer] = useState(null);

  useEffect(() => {
    fetch(config.BASE_PATH + "/hello")
      .then((response) => response.text())
      .then((error) => {
        setMessage(error);
      });
  }, []);

  const onSubmit = (ev) => {
    ev.preventDefault();
    setAnswer(rot13(someText));
  };

  return (
    <div className="App">
      <header className="App-header">
        <img
          src={process.env.PUBLIC_URL + "/assets/logo.svg"}
          className="App-logo"
          alt="logo"
        />
        <h1 className="App-title">{message}</h1>
      </header>
      <p className="App-intro">
        To get started, edit <code>src/App.js</code> and save to reload.
      </p>
      <p>
        <Link to="test">App route</Link>
      </p>
      <form action="" onSubmit={onSubmit}>
        <p>
          <label htmlFor="some-text">Input some text: </label>
          <input
            name="some-text"
            id="some-text"
            value={someText}
            onChange={(ev) => setSomeText(ev.currentTarget.value)}
          />
        </p>
        <button>Calculate</button>
      </form>
      {answer && (
        <p>
          The answer is: <strong>{answer}</strong>
        </p>
      )}
    </div>
  );
};

export default Home;
