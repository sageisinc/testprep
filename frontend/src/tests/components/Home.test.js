import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Home from "../../modules/app/components/Home";

describe("Home", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Home />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calculates the value as expected", async () => {
    render(
      <MemoryRouter>
        <Home />
      </MemoryRouter>
    );

    const inputText = screen.getByLabelText(/input some text/i);
    fireEvent.change(inputText, {
      target: { value: "Sreenzragnf qr qrfraibyirzragb" },
    });
    const calculateButton = screen.getByText(/calculate/i);
    fireEvent.click(calculateButton);
    await waitFor(() => {
      screen.getByText(/Ferramentas de desenvolvemento/);
    });
  });
});
