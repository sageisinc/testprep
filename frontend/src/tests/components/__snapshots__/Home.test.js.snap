// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="/assets/logo.svg"
    />
    <h1
      className="App-title"
    />
  </header>
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <p>
    <a
      href="/test"
      onClick={[Function]}
    >
      App route
    </a>
  </p>
  <form
    action=""
    onSubmit={[Function]}
  >
    <p>
      <label
        htmlFor="some-text"
      >
        Input some text: 
      </label>
      <input
        id="some-text"
        name="some-text"
        onChange={[Function]}
        value="Sreenzragnf"
      />
    </p>
    <button>
      Calculate
    </button>
  </form>
</div>
`;
